components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -3127,24 +3052,88 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
 
         const user = await this.checkAndBlockUser("deleteOrgAuthProvider");
 
+        // check for "orgGitAuthProviders" feature flag
         const team = await this.getTeam(ctx, params.organizationId);
         if (!team) {
             throw new ApplicationError(ErrorCodes.BAD_REQUEST, "Invalid organizationId");
         }
-
         await this.guardWithFeatureFlag("orgGitAuthProviders", user, team.id);
 
         await this.guardTeamOperation(params.organizationId || "", "update");
-        await this.auth.checkPermissionOnOrganization(user.id, "write_git_provider", params.organizationId);
 
-        // Find the matching auth provider we're attempting to delete
-        const orgProviders = await this.authProviderService.getAuthProvidersOfOrg(team.id);
-        const authProvider = orgProviders.find((p) => p.id === params.id && p.organizationId === params.organizationId);
+        await this.authProviderService.deleteAuthProviderOfOrg(user.id, params.organizationId, params.id);
+    }
+
+    async getAuthProvider(ctx: TraceContextWithSpan, id: string): Promise<AuthProviderEntry> {
+        traceAPIParams(ctx, { id });
+
+        const user = await this.checkAndBlockUser("getAuthProvider");
+
+        const result = await this.authProviderService.getAuthProvider(user.id, id);
+        if (!result) {
+            throw new ApplicationError(ErrorCodes.NOT_FOUND, "Provider resource not found.");
+        }
+        return AuthProviderEntry.redact(result);
+    }
+
+    /**
+     * Delegates to `deleteOrgAuthProvider` or `deleteOwnAuthProvider` depending on the ownership
+     * of the specified auth provider.
+     */
+    async deleteAuthProvider(ctx: TraceContextWithSpan, id: string): Promise<void> {
+        traceAPIParams(ctx, { id });
+
+        const user = await this.checkAndBlockUser("deleteAuthProvider");
+
+        const authProvider = await this.authProviderService.getAuthProvider(user.id, id);
```

Comment:
does it mean that for delete and update we need to look up the same query twice?

Reply:
That's the mentioned quirks with user-level vs. org-level providers. We've copied the json-rpc API method in the past, and now we want to introduce a single RPC which internally needs to find the provider by id before it can delegate. 

The downside of having two reads here would remain til we sunset the user-level providers.

I'd prefer this to the alternatives:
1. having the additional find-by-id call in frontend, which would come with more latency.
2. having separate RPCs to repeat the quirks in the new API, which would come with a breaking change later on.

Reply:
This API seems to be infrequently used so all right for now. Let's add TODO in the code?

Reply:
roger that.